src/torrent.cpp

```diff
@@ -1138,6 +1138,20 @@ bool is_downloading_state(int const st)
 		if (m_peer_list) m_peer_list->clear();
 	}
 
+	void torrent::set_sequential_range(piece_index_t start_piece, piece_index_t end_piece)
+	{
+		if (!has_picker()) {
+			if (!valid_metadata() || !m_connections_initialized) return;
+			else if (!m_have_all
+				|| settings().get_int(settings_pack::suggest_mode)
+				== settings_pack::suggest_read_cache)
```

Comment:
why is `suggest_mode` a condition here?

Reply:
That's what the condition was (among other things) in need_picker. You mentioned to me to look at that in order to copy its set of conditions here.